src/cpp/ext/otel/otel_plugin.cc

```diff
@@ -80,8 +65,32 @@ absl::flat_hash_set<std::string> BaseMetrics() {
       std::string(grpc::OpenTelemetryPluginBuilder::
                       kServerCallRcvdTotalCompressedMessageSizeInstrumentName)};
 }
+
+// TODO(yashykt): This should go away when we move to the
+// GlobalStatsPluginRegistry.
+std::atomic<bool> g_otel_plugin_enabled(false);
+
 }  // namespace
 
+// TODO(yashykt): Extend this to allow multiple OpenTelemetry plugins to be
```

Comment:
I think we are on the same page about this.

The `g_otel_plugin_state_` is an existing thing that will go away once we have the GlobalStatsPluginRegistry. Currently, the limitation is that we can have only once otel stats plugin enabled, so I think it's fine to just have a single global that works for the otel stats plugin.

If we do this in the CSM plugin's target and server selector callbacks, it will just disable the CSM label addition, but the base OTel functionality will continue which I don't think we want.

Reply:
Done as discussed on chat